### PR TITLE
[FLINK-11515][client] Port ClientTest

### PR DESCRIPTION
**Pull request by zentol**
## What is the purpose of the change

This PR updates the `ClientTest` to not rely on legacy code.

## Brief change log

* replace actor system with a `MiniClusterResource`
* replace StandaloneClusterClient with MiniClusterClient
* move `shouldSubmitToJobClientFails()` to `RestClusterClientTest`

### CODE REVIEW

File: flink-clients/src/test/java/org/apache/flink/client/program/ClientTest.java [hunk 1]
```diff
@@ -75,12 +63,14 @@
  */
 public class ClientTest extends TestLogger {
 
+	@ClassRule
+	public static final MiniClusterResource MINI_CLUSTER_RESOURCE =
+		new MiniClusterResource(new MiniClusterResourceConfiguration.Builder().build());
+
```
Comment (TisonKun) [inline flink-clients/src/test/java/org/apache/flink/client/program/ClientTest.java]: Cannot find a call to `MINI_CLUSTER_RESOURCE.before()`. Is it intended?

Comment (zentol) [reply flink-clients/src/test/java/org/apache/flink/client/program/ClientTest.java]: `@ClassRule` does that behind the scenes.

File: flink-clients/src/test/java/org/apache/flink/client/program/ClientTest.java [hunk 2]
```diff
@@ -102,39 +92,14 @@ public void setUp() throws Exception {
...
 	/**
 	 * Tests that invalid detached mode programs fail.
 	 */
 	@Test
 	public void testDetachedMode() throws Exception{
-		jobManagerSystem.actorOf(
-			Props.create(SuccessReturningActor.class),
-			JobMaster.JOB_MANAGER_NAME);
-		StandaloneClusterClient out = new StandaloneClusterClient(config);
+		final ClusterClient<?> out = new MiniClusterClient(new Configuration(), MINI_CLUSTER_RESOURCE.getMiniCluster());
```
Comment (GJL) [inline flink-clients/src/test/java/org/apache/flink/client/program/ClientTest.java]: It was called `out` before but I don't understand why. I think `clusterClient` is a better name.

File: flink-clients/src/test/java/org/apache/flink/client/program/ClientTest.java [hunk 3]
```diff
@@ -216,41 +177,13 @@ public void shouldSubmitToJobClient() throws Exception {
...
-		}
-	}
-
 	/**
 	 * This test verifies that the local execution environment cannot be created when
 	 * the program is submitted through a client.
 	 */
 	@Test
 	public void tryLocalExecution() {
 		try {
```
Comment (GJL) [inline flink-clients/src/test/java/org/apache/flink/client/program/ClientTest.java]: Maybe delete the 
```
catch (Exception e) {
			e.printStackTrace();
			fail(e.getMessage());
		}
``` 

block here as well

File: flink-clients/src/test/java/org/apache/flink/client/program/ClientTest.java [hunk 4]
```diff
@@ -280,10 +213,6 @@ public Void answer(InvocationOnMock invocation) throws Throwable {
 	@Test
 	public void testGetExecutionPlan() {
 		try {
```
Comment (GJL) [inline flink-clients/src/test/java/org/apache/flink/client/program/ClientTest.java]: Maybe delete the 
```
catch (Exception e) {
			e.printStackTrace();
			fail(e.getMessage());
		}
``` 

block here as well

File: flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java [hunk 5]
```diff
@@ -752,6 +752,33 @@ public void testRetriableSendOperationIfConnectionErrorOrServiceUnavailable() th
...
+	}
+
+	private final class SubmissionFailingHandler extends TestHandler<JobSubmitRequestBody, JobSubmitResponseBody, EmptyMessageParameters> {
+
+		private SubmissionFailingHandler() {
+			super(JobSubmitHeaders.getInstance());
+		}
+
+		@Override
+		protected CompletableFuture<JobSubmitResponseBody> handleRequest(@Nonnull HandlerRequest<JobSubmitRequestBody, EmptyMessageParameters> request, @Nonnull DispatcherGateway gateway) throws RestHandlerException {
```
Comment (GJL) [inline flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java]: Line maybe too long.

**Pull request merged**